rllib/utils/tests/test_checkpoint_utils.py

```diff
@@ -32,44 +41,164 @@ def test_get_checkpoint_info_v0_1(self):
+                os.makedirs(pol1_dir)
+                pol2_dir = os.path.join(checkpoint_dir, "policies", "pol2")
+                os.makedirs(pol2_dir)
+                # policy_state.pkl
+                Path(os.path.join(pol1_dir, "policy_state.pkl")).touch()
+                Path(os.path.join(pol2_dir, "policy_state.pkl")).touch()
 
-            info = get_checkpoint_info(checkpoint_dir)
-            self.assertTrue(info["type"] == "Algorithm")
-            self.assertTrue(str(info["checkpoint_version"]) == "1.0")
-            self.assertTrue(info["checkpoint_dir"] == checkpoint_dir)
-            self.assertTrue(info["state_file"] == algo_state_file)
-            self.assertTrue(
-                "pol1" in info["policy_ids"] and "pol2" in info["policy_ids"]
-            )
+                info = get_checkpoint_info(checkpoint_dir)
+                self.assertTrue(info["type"] == "Algorithm")
+                self.assertTrue(str(info["checkpoint_version"]) == "1.1")
+                self.assertTrue(info["checkpoint_dir"] == checkpoint_dir)
+                self.assertTrue(info["state_file"] == algo_state_file)
+                self.assertTrue(
+                    "pol1" in info["policy_ids"] and "pol2" in info["policy_ids"]
+                )
 
-    def test_get_policy_checkpoint_info_v1_0(self):
+    def test_get_policy_checkpoint_info_v1_1(self):
         # Create a simple (dummy) v1.0 Policy checkpoint.
         with tempfile.TemporaryDirectory() as checkpoint_dir:
-            # Old checkpoint-[iter] file.
             policy_state_file = os.path.join(checkpoint_dir, "policy_state.pkl")
             Path(policy_state_file).touch()
 
             info = get_checkpoint_info(checkpoint_dir)
             self.assertTrue(info["type"] == "Policy")
-            self.assertTrue(str(info["checkpoint_version"]) == "1.0")
+            self.assertTrue(str(info["checkpoint_version"]) == "1.1")
             self.assertTrue(info["checkpoint_dir"] == checkpoint_dir)
             self.assertTrue(info["state_file"] == policy_state_file)
             self.assertTrue(info["policy_ids"] is None)
 
+    def test_msgpack_checkpoint_translation(self):
+        """Tests, whether a checkpoint can be translated into a msgpack-checkpoint ...
+
+        ... and recovered back into and Algorithm, which is identical to a
+        pickle-checkpoint-recovered Algorithm (given same initial config).
+        """
+        # Base config used for both pickle-based checkpoint and msgpack-based one.
+        config = SimpleQConfig().environment("CartPole-v1")
+        # Build algorithm object.
+        algo1 = config.build()
+        # Get its state.
+        pickle_state = algo1.__getstate__()
+        # Create standard (pickle-based) checkpoint.
+        with tempfile.TemporaryDirectory() as pickle_cp_dir:
+            pickle_cp_dir = algo1.save(checkpoint_dir=pickle_cp_dir)
+            # Now convert pickle checkpoint to msgpack using the provided
```

Comment:
can we check that the pickle checkpoint is definitely different than the non pickle checkpoint. We could use version number for example, but also inspect the checkpoint json file further.

Reply:
Added additional checks (via get_checkpoint_info, which reads the JSON info data from the checkpoints).